modules/service/src/test/scala/lucuma/odb/graphql/issue/shortcut/2772.scala

```diff
@@ -0,0 +1,275 @@
+import lucuma.core.model.sequence.StepConfig.Gcal
+import lucuma.core.util.TimeSpan
+import lucuma.odb.graphql.enums.Enums
+import lucuma.odb.graphql.query.ObservingModeSetupOperations
+import lucuma.odb.service.Services
+import lucuma.odb.smartgcal.data.Gmos.GratingConfigKey
+import lucuma.odb.smartgcal.data.Gmos.TableKey
+import lucuma.odb.smartgcal.data.Gmos.TableRow
+import lucuma.odb.smartgcal.data.SmartGcalValue
+import lucuma.odb.smartgcal.data.SmartGcalValue.LegacyInstrumentConfig
+import natchez.Trace.Implicits.noop
+import skunk.Session
+
+class ShortCut_2772 extends OdbSuite with ObservingModeSetupOperations {
+
+  val user: User = TestUsers.service(3)
+  override val validUsers: List[User] = List(user, user)
+  val createProgram: IO[Program.Id] = createProgramAs(user, "ShortCut 2772")
+
+  // R831 grating with a 1" slit and IQ=1.0"
+
+  override def dbInitialization: Option[Session[IO] => IO[Unit]] = Some { s =>
+    val tableRow1: TableRow.North =
+      TableRow(
+        PosLong.unsafeFrom(1),
+        TableKey(
+          GratingConfigKey(
+            GmosNorthGrating.R831_G5302,
+            GmosGratingOrder.One,
+            BoundedInterval.unsafeOpenUpper(Wavelength.Min, Wavelength.Max)
+          ).some,
+          none,
+          GmosNorthFpu.LongSlit_1_00.some,
+          GmosXBinning.Two,
+          GmosYBinning.Four,
+          GmosAmpGain.Low
+        ),
+        SmartGcalValue(
+          Gcal(
+            Gcal.Lamp.fromContinuum(GcalContinuum.QuartzHalogen5W),
+            GcalFilter.Gmos,
+            GcalDiffuser.Ir,
+            GcalShutter.Open
+          ),
+          GcalBaselineType.Night,
+          PosInt.unsafeFrom(1),
+          LegacyInstrumentConfig(
+            TimeSpan.unsafeFromMicroseconds(1_000_000L)
+          )
+        )
+      )
+
+    Enums.load(s).flatMap { e =>
+      val services = Services.forUser(user /* doesn't matter*/, e)(s)
+      services.transactionally {
+        services.smartGcalService.insertGmosNorth(1, tableRow1)
+      }
+    }
+  }
+
+  def createTargetWithProfile(pid: Program.Id): IO[Target.Id] =
+    query(
+      user  = user,
+      query =
+      s"""
+         mutation {
+           createTarget(input: {
+             programId: ${pid.asJson},
+             SET: {
+               name: "V1647 Orionis"
+               sidereal: {
+                 ra: { hms: "05:46:13.137" },
+                 dec: { dms: "-00:06:04.89" },
+                 epoch: "J2000.0",
+                 properMotion: {
+                   ra: {
+                     milliarcsecondsPerYear: 0.918
+                   },
+                   dec: {
+                     milliarcsecondsPerYear: -1.057
+                   },
+                 },
+                 radialVelocity: {
+                   kilometersPerSecond: 27.58
+                 },
+                 parallax: {
+                   milliarcseconds: 2.422
+                 }
+               },
+               sourceProfile: {
```

Comment:
Did you say the bug  was related  to the source profiile

Reply:
That's correct.  For a point source, we use the smaller of the IQ and the slit width to determine the object size.  For uniform, it is effectively just the slit width.  For Gaussian, it is the explicit FWHM.  The issue was that the conversion of the IQ to the corresponding angle was incorrect.  For example an IQ of `ONE_POINT_ZERO` was turned into an angle of 10 microarcseconds instead of 1 arcsecond.  This [was fixed in core](https://github.com/gemini-hlsw/lucuma-core/pull/885).